generated/nifgen/nifgen_service.cpp

```diff
@@ -0,0 +1,3147 @@
+    }
+  }
+
+  //---------------------------------------------------------------------
+  //---------------------------------------------------------------------
+  ::grpc::Status NiFgenService::ClearFreqList(::grpc::ServerContext* context, const ClearFreqListRequest* request, ClearFreqListResponse* response)
+  {
+    if (context->IsCancelled()) {
+      return ::grpc::Status::CANCELLED;
+    }
+    try {
+      auto vi_grpc_session = request->vi();
+      ViSession vi = session_repository_->access_session(vi_grpc_session.id(), vi_grpc_session.name());
+      ViInt32 frequency_list_handle = request->frequency_list_handle();
+      auto status = library_->ClearFreqList(vi, frequency_list_handle);
+      response->set_status(status);
+      return ::grpc::Status::OK;
+    }
+    catch (nidevice_grpc::LibraryLoadException& ex) {
+      return ::grpc::Status(::grpc::NOT_FOUND, ex.what());
+    }
+  }
+
+  //---------------------------------------------------------------------
+  //---------------------------------------------------------------------
+  ::grpc::Status NiFgenService::ClearInterchangeWarnings(::grpc::ServerContext* context, const ClearInterchangeWarningsRequest* request, ClearInterchangeWarningsResponse* response)
+  {
+    if (context->IsCancelled()) {
+      return ::grpc::Status::CANCELLED;
+    }
+    try {
+      auto vi_grpc_session = request->vi();
+      ViSession vi = session_repository_->access_session(vi_grpc_session.id(), vi_grpc_session.name());
+      auto status = library_->ClearInterchangeWarnings(vi);
+      response->set_status(status);
+      return ::grpc::Status::OK;
+    }
+    catch (nidevice_grpc::LibraryLoadException& ex) {
+      return ::grpc::Status(::grpc::NOT_FOUND, ex.what());
+    }
+  }
+
+  //---------------------------------------------------------------------
+  //---------------------------------------------------------------------
+  ::grpc::Status NiFgenService::ClearUserStandardWaveform(::grpc::ServerContext* context, const ClearUserStandardWaveformRequest* request, ClearUserStandardWaveformResponse* response)
+  {
+    if (context->IsCancelled()) {
+      return ::grpc::Status::CANCELLED;
+    }
+    try {
+      auto vi_grpc_session = request->vi();
+      ViSession vi = session_repository_->access_session(vi_grpc_session.id(), vi_grpc_session.name());
+      ViConstString channel_name = request->channel_name().c_str();
+      auto status = library_->ClearUserStandardWaveform(vi, channel_name);
+      response->set_status(status);
+      return ::grpc::Status::OK;
+    }
+    catch (nidevice_grpc::LibraryLoadException& ex) {
+      return ::grpc::Status(::grpc::NOT_FOUND, ex.what());
+    }
+  }
+
+  //---------------------------------------------------------------------
+  //---------------------------------------------------------------------
+  ::grpc::Status NiFgenService::Close(::grpc::ServerContext* context, const CloseRequest* request, CloseResponse* response)
```

Comment:
Something is wrong with `Close` and `CloseExtCal` codegen here. They should be calling `      session_repository_->remove_session(vi);` [like this](https://github.com/doshirohan/grpc-device/blob/697fa14a28fdea49ae9ac8d39d0f49139a64379c/generated/nidcpower/nidcpower_service.cpp#L1170). I think you might be missing some tag. Check with Kruthi/Akshata about this.

Reply:
Yes, I think adding close methods in the config.py step has been missed.

Reply:
For CloseExtCal, you need to add tag like [this](https://github.com/doshirohan/grpc-device/blob/main/source/codegen/metadata/nidcpower/functions.py#L1040):
'custom_close_method': True,

For all initialize calibration functions, you need to add tag similar to [this](https://github.com/doshirohan/grpc-device/blob/main/source/codegen/metadata/nidcpower/functions.py#L2515)

For the normal Close function, I think when we add the corresponding initialize method and tag that with init method, then it might generate properly. @Akshata440  can you confirm?

Reply:
Yes, that is correct @kruthi-NI

Reply:
I could only add the init_method and custom_close to the InitExtCal. When I added for the other calibration methods, I was getting a build failure.

Reply:
I think you only need init_method on `InitExtCal`, `InitWithOptions` and `InitializeWithChannels`. You only need the init_method tag on things creating a ViSession (i.e., having ViSession as an out paramater). For custom_close, I believe you only need it for `InitExtCal` as the other Init's should just be calling the default `Close`.

Reply:
Done. Added init_method to InitExtCal